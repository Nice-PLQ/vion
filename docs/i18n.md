# 国际化

Vion 默认采用中文语言，如果需要切换其他语言，请参考如下

### 1、切换语言

Vion 通过 Locale 组件实现多语言支持，使用`Locale.use`方法切换语言环境，比如切换到英文环境

```js
// 按需引入
import { createApp } from "vue";
import { ActionSheet, Locale } from "vion";
import messages from "vion/es/locale/lang/en-US";

Locale.use("en-US", messages);

createApp()
  .use(ActionSheet);
  .use(Locale);

```

或

```js
// 全量引入Vion
import { createApp } from "vue";
import Vion from "vion";
import messages from "vion/es/locale/lang/en-US";

vion.Locale.use("en-US", messages);

createApp().use(Vion);
```

### 2、语言包

目前 vion 内置了以下几种语言

- 简体中文（zh-CN）
- 繁体中文（zh-Hant）
- 英语（en-US）
- 日语（ja-JP）

如果不能满足业务场景，只需要按照[语言包格式](https://github.com/Nice-PLQ/vion/blob/master/src/locale/lang/zh-CN.js)自定义语言，然后使用`Locale`组件加载即可，如：

```js
import messages from "./lang/fr-FR"; // 自定义的【法语】语言包
Locale.use("fr-FR", messages);
```

### 3、引入多个语言包

```js
import frMessages from "./lang/fr-FR"; // 自定义的【法语】语言包
import trMessages from "./lang/tr-TR"; // 自定义的【土耳其语】语言包

Locale.add("tr-TR", trMessages); // 将语言包加入到集合中
Locale.use("fr-FR", frMessages); // 默认使用法语作为首选语言
```
