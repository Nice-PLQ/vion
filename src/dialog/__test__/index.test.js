import { mount } from '@vue/test-utils';
import { next } from '../../../test';
import Dialog from '../dialog.vue';
import i18n from '../../locale';

describe('Dialog', () => {
  test('弹窗渲染', async () => {
    const wrapper = mount(Dialog, {
      props: {
        title: '标题',
        content: '内容',
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $dialog: Dialog,
            $locale: i18n,
          }
        }
      }
    });

    expect(wrapper.find('.v-dialog__title').text()).toBe('标题');
    expect(wrapper.find('.v-dialog__description').text()).toBe('内容');

    wrapper.setProps({ visible: true });
    await next();
    expect(wrapper.find('.v-popup__content').isVisible()).toBe(true);
    expect(wrapper.find('.v-dialog').element).toMatchSnapshot();
  });

  test('slot:default', () => {
    const wrapper = mount(Dialog, {
      slots: {
        default: '<span>内容</span>'
      },
      props: {
        visible: true,
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $dialog: Dialog,
            $locale: i18n,
          }
        }
      }
    });

    expect(wrapper.find('.v-dialog__description span').html()).toBe('<span>内容</span>');
  });

  test('弹窗按钮操作', () => {
    const fn = jest.fn();
    const wrapper = mount(Dialog, {
      props: {
        title: '标题',
        content: '内容',
        visible: true,
        callback: fn,
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $dialog: Dialog,
            $locale: i18n,
          }
        }
      }
    });

    wrapper.findAll('.v-dialog__btn')[0].trigger('click');
    expect(wrapper.emitted('cancel')).toBeTruthy();
    expect(fn).toHaveBeenCalled();
  });

  test('beforeClose: 关闭前的回调', () => {
    const fn = jest.fn();
    const wrapper = mount(Dialog, {
      props: {
        title: '标题',
        content: '内容',
        visible: true,
        beforeClose: fn,
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $dialog: Dialog,
            $locale: i18n,
          }
        }
      }
    });

    wrapper.findAll('.v-dialog__btn')[0].trigger('click');
    expect(wrapper.emitted('cancel')).toBeTruthy();
    expect(fn).toHaveBeenCalled();
  });

  test('自定义选择按钮', () => {
    const wrapper = mount(Dialog, {
      props: {
        title: '标题',
        content: '内容',
        visible: true,
        buttons: ['使用', '不使用', { text: '取消', color: '#FF596A' }],
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $dialog: Dialog,
            $locale: i18n,
          }
        }
      }
    });

    wrapper.find('.v-dialog__btn').trigger('click');
    expect(wrapper.emitted('click')[0][0]).toBe(0);
  });
});
