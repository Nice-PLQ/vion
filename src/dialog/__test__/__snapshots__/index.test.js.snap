// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog 弹窗渲染 1`] = `
<div
  class="v-popup v-dialog"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      aria-label="关闭"
      class="v-popup__mask"
      role="button"
      style=""
    />
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="false"
  >
    <!--v-if-->
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade-zoom"
    persisted="true"
  >
    <div
      class="v-popup__content v-popup__content_center"
      style=""
    >
      
      <div
        class="v-dialog__box"
      >
        <div
          class="v-dialog__content"
        >
          <div
            class="v-dialog__title"
          >
            标题
          </div>
          <div
            class="v-dialog__description"
          >
            
            <div>
              内容
            </div>
            
          </div>
        </div>
        <div
          class="v-dialog__footer"
        >
          <button
            class="v-dialog__btn"
          >
            取消
          </button>
          <button
            class="v-dialog__btn"
          >
            确定
          </button>
        </div>
      </div>
      
    </div>
  </transition-stub>
</div>
`;
